MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java

```diff
@@ -1529,39 +1397,31 @@ private void maybeShowPopup(MouseEvent e) {
             /*
              * if(!person.isAssigned()) { cbMenuItem.setSelected(true); }
              */
-            if (oneSelected && person.isActive()
-                    && !(person.isPrisoner() || person.isBondsman())) {
+            if (oneSelected && person.isActive() && !(person.isPrisoner() || person.isBondsman())) {
                 for (Unit unit : gui.getCampaign().getUnits()) {
                     if (!unit.isAvailable()) {
                         continue;
                     }
                     if (unit.usesSoloPilot()) {
-                        if (unit.canTakeMoreDrivers()
-                                && person.canDrive(unit.getEntity())
+                        if (unit.canTakeMoreDrivers() && person.canDrive(unit.getEntity())
                                 && person.canGun(unit.getEntity())) {
-                            cbMenuItem = new JCheckBoxMenuItem(
-                                    unit.getName());
-                            // TODO: check the box
+                            cbMenuItem = new JCheckBoxMenuItem(unit.getName());
                             cbMenuItem.setActionCommand(makeCommand(CMD_ADD_PILOT, unit.getId().toString()));
                             cbMenuItem.addActionListener(this);
                             pilotMenu.add(cbMenuItem);
                         }
                     } else if (unit.usesSoldiers()) {
-                        if (unit.canTakeMoreGunners()
-                                && person.canGun(unit.getEntity())) {
-                            cbMenuItem = new JCheckBoxMenuItem(
-                                    unit.getName());
-                            // TODO: check the box
+                        if (unit.canTakeMoreGunners() && person.canGun(unit.getEntity())) {
+                            cbMenuItem = new JCheckBoxMenuItem(unit.getName());
+                            cbMenuItem.setSelected(person.getUnitId() == unit.getId());
```

Comment:
Is reference equality what we want here?

Reply:
If the person is assigned to this unit, we want it selected. A comparison of the unit id should do that perfectly. I'm pretty sure it is correct, unless I'm missing something.

Reply:
You're currently testing if the person's unit ID points to the same location in memory as the unit's ID. You likely intended this to compare the **value** of the IDs, right? Outside of `int`, `boolean`, `float`, and other primitive types, you should be using `a.equals(b)` or `Objects.equals(a, b)`.

Reply:
Fixed